src/tscore/unit_tests/test_IntrusiveHashMap.cc

```diff
@@ -167,12 +169,12 @@ TEST_CASE("IntrusiveHashMapManyStrings", "[IntrusiveHashMap]")
   strings.reserve(N);
   for (int i = 0; i < N; ++i) {
     auto len = length_gen(randu);
-    char *s  = static_cast<char *>(malloc(len + 1));
+    std::string s;
```

Comment:
Given the maximum length of the string is known, have a static buffer of sufficient length, put the generated string there, and then `emplace_back(buff, len)`.

Reply:
I am doing what the test originally had, creating a vector of strings with random lengths and values.